securedrop/models.py

```diff
@@ -294,18 +295,16 @@ def __repr__(self) -> str:
         return '<Reply %r>' % (self.filename)
 
     def to_json(self) -> 'Dict[str, Any]':
-        journalist_username = "deleted"
-        journalist_first_name = ""
-        journalist_last_name = ""
-        journalist_uuid = "deleted"
-        if self.journalist:
-            journalist_username = self.journalist.username
-            journalist_first_name = self.journalist.first_name
-            journalist_last_name = self.journalist.last_name
+        journalist_username = self.journalist.username
+        journalist_first_name = self.journalist.first_name or ""
+        journalist_last_name = self.journalist.last_name or ""
+        if journalist_username == "deleted":
+            # special-case the "deleted" user
+            journalist_uuid = "deleted"
```

Comment:
I think this will break the API contract, as it's not a valid uuid. It will require clients to implement custom parsing logic just for this field. I would recommend returning a valid uuid instead, and adding an explicit boolean field, such as is_special_deleted_journalist.

Reply:
This is matching the current behavior of the API, which returns "deleted" for the UUID of a null journalist. I agree that it should return a real UUID instead as I hinted in my PR message, but AIUI it needs coordination with the workstation.

Reply:
It makes sense to me to go ahead and use a valid UUID. Downstream, the client-side database has this field:
```py
class User(Base):
    ...
    def deleted(self) -> bool:
        return True if self.uuid == "deleted" else False
```

Which can be updated and released, before your changes, to:
```py
class User(Base):
    ...
    def deleted(self) -> bool:
        return True if self.username == "deleted" else False
```

The reason we didn't call this something like `is_special_deleted_journalist` was because we imagined potentially adding support to distinguish replies sent by distinct deleted journalists. But the non-null and unique username presents a challenge for the case where an org wants to distinguish replies sent by distinct deleted journalists without having to freeze usernames or create unique username replacements, e.g. `deleted_0` `deleted_1` `deleted_2` ... and so on. For now, just supporting one deleted user account that everything gets rolled up into is a big improvement.

Reply:
You can test against https://github.com/freedomofpress/securedrop-client/tree/users-endpoint if it helps.

Reply:
Done.